app/src/main/java/com/aptoide/uploader/MainPresenter.kt

```diff
@@ -23,11 +25,16 @@ class MainPresenter(val view: MainView, val accountManager: AptoideAccountManage
 
   override fun present() {
     compositeDisposable = CompositeDisposable()
-
+    handleLoginView()
     handleIntentEvents()
     onDestroyDisposeComposite()
   }
 
+  private fun handleLoginView() {
+    autoLoginManager.firstStoredUserCredentials
+    autoLoginManager.checkLoginStatus(mainNavigator)
```

Comment:
this name hides the implementation behaviour (navigation). Calling a method that looks like it is going to do some sort of validation but in the end it will actually do navigation is strange. The next person that sees this might not understand where the navigation is.

Reply:
Yeah i agree. The problem with that method is that it does more than just one thing ate once (check where to navigate and with what properties to show in the view) but I'll rename it to checkAvailableFieldsAndNavigateTo